### PR TITLE
Min Gap regulatory element

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Add minimum gap relation element

## Motivation and Context
Road weather use case

## How Has This Been Tested?
Local unit test

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: common/lanelet2_extension/docs/RegulatoryElements.md [hunk 1]
```diff
@@ -2,6 +2,53 @@
...
+| **Role** | **Possible Type** | **description**                |
+|-------------|--------------|----------------------------------|
+| **refers**    | **Lanelet, Area**    | The region of the roadway this minimum gap applies to |
+
+### Custom Attributes
+
+| **Key** | **Value Type** | **description**                |
+|-------------|--------------|----------------------------------|
+| **subtype** | **digital_minimun_gap**    | Subtype name |
+| **mingap** | **double**    | The minimum gap to set. In an osm file units must be one of the following m/s, mps, km/h, kmh, m/h, mph |
```
Comment (msmcconnell) [inline common/lanelet2_extension/docs/RegulatoryElements.md]: Units should be meters right?

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
+#pragma once
+/*
+ * Copyright (C) 2020 LEIDOS.
```
Comment (msmcconnell) [inline common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h]: 2021

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  /**
+   * @brief Returns true if the given participant must follow this minimum gap 
+   *
+   * @return True if this minimum gap  should apply to the given participant
+   */
+  bool appliesTo(const std::string& participant) const;
+
+  /**
+   * @brief Static helper function that creates a minimum gap  based on the provided double, start, end lines, and the
```
Comment (msmcconnell) [inline common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h:73]: start and end lines doesn't seem to refer to anything here. Did you mean provided min_gap and lanelets?

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+   * affected participants
+   *
+   * @param id The lanelet::Id of this object
+   * @param min_gap The double which will be treated as the minimum gap  in this region
```
Comment (msmcconnell) [inline common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h]: Specify units here. (meters)

File: common/lanelet2_extension/lib/DigitalMinimumGap.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
+#pragma once
+/*
+ * Copyright (C) 2020 LEIDOS.
```
Comment (msmcconnell) [inline common/lanelet2_extension/lib/DigitalMinimumGap.cpp]: 2021

File: common/lanelet2_extension/test/src/DigitalMinimumGapTest.cpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019-2021 LEIDOS.
```
Comment (msmcconnell) [inline common/lanelet2_extension/test/src/DigitalMinimumGapTest.cpp]: Just 2021

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h (hunk 3 above)
Comment (dan-du-car) [reply common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h:73]: yes

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h (hunk 4 above)
Comment (dan-du-car) [reply common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h]: https://github.com/usdot-fhwa-stol/autoware.ai/pull/145/commits/0e06e0075d392abb106f5857b323653029c644ed

File: common/lanelet2_extension/test/src/DigitalMinimumGapTest.cpp (hunk 6 above)
Comment (dan-du-car) [reply common/lanelet2_extension/test/src/DigitalMinimumGapTest.cpp]: https://github.com/usdot-fhwa-stol/autoware.ai/pull/145/commits/91a4c828df90fdfcf4dbfb864c1640a07bb8a54d

File: common/lanelet2_extension/lib/DigitalMinimumGap.cpp (hunk 5 above)
Comment (dan-du-car) [reply common/lanelet2_extension/lib/DigitalMinimumGap.cpp]: https://github.com/usdot-fhwa-stol/autoware.ai/pull/145/commits/91a4c828df90fdfcf4dbfb864c1640a07bb8a54d

File: common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h (hunk 2 above)
Comment (dan-du-car) [reply common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/DigitalMinimumGap.h]: https://github.com/usdot-fhwa-stol/autoware.ai/pull/145/commits/8f6c38d08ae86e738a2bc7aaa9a91ccc1a60b02c

File: common/lanelet2_extension/docs/RegulatoryElements.md (hunk 1 above)
Comment (dan-du-car) [reply common/lanelet2_extension/docs/RegulatoryElements.md]: https://github.com/usdot-fhwa-stol/autoware.ai/pull/145/commits/8f6c38d08ae86e738a2bc7aaa9a91ccc1a60b02c

**Review by msmcconnell — APPROVED** [review]

**Comment by msmcconnell** [comment]
This PR is needed for https://github.com/usdot-fhwa-stol/carma-platform/issues/1205

**Pull request merged**